### PR TITLE
refactor(notification): refactor component to use hooks

### PR DESCRIPTION
**Pull request by aledavila**
Refactoring Notification component to use hooks

#### Changelog

**New**

- added proptypes to NotificationIcon

**Changed**

- refactored `NotificationIcon` return statement
- refactored icontype function
- refactored component to use hooks
- updated tests to account for no longer using `state`

#### Testing / Reviewing

Make sure notification is rendering.

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

File: packages/react/src/components/Notification/Notification.js [hunk 1]
```diff
@@ -20,354 +20,411 @@ import Button from '../Button';
...
+  onCloseButtonClick, // eslint-disable-line
+  iconDescription, // eslint-disable-line
+  className,
+  caption,
+  subtitle,
+  title,
+  kind,
+  lowContrast,
+  hideCloseButton,
+
```
Comment (tw15egan) [inline packages/react/src/components/Notification/Notification.js]: Can we remove the spaces?

File: packages/react/src/components/Notification/Notification.js [hunk 2]
```diff
@@ -20,354 +20,411 @@ import Button from '../Button';
...
+  children,
+  timeout,
+
```
Comment (tw15egan) [inline packages/react/src/components/Notification/Notification.js]: See above

File: packages/react/src/components/Notification/Notification.js [hunk 3]
```diff
@@ -20,354 +20,411 @@ import Button from '../Button';
...
+  notificationType,
+  onCloseButtonClick, // eslint-disable-line
+  iconDescription, // eslint-disable-line
+  className,
+  subtitle,
+  title,
+  kind,
+  lowContrast,
+  hideCloseButton,
+
```
Comment (tw15egan) [inline packages/react/src/components/Notification/Notification.js]: See above

**Review by tw15egan — APPROVED** [review]
Looks good to me, just tiny style fixes 👍

**Comment by aledavila** [comment]
@tw15egan done! thanks

**Pull request merged**